### PR TITLE
Qt: Add "Big Picture" UI

### PR DESCRIPTION
**Pull request by stenzek**
### Description of Changes

This adds the "big picture" mode previously seen in the RetroAchievements PR to PCSX2. Mostly complete now, ready for testing.

Note: You can launch it in Qt via the menu/toolbar, or via `-bigpicture` on the command line.

### Rationale behind Changes

Making PCSX2 usable only with a controller.

### Suggested Testing Steps

Test new interface.

### CODE REVIEW

**Comment by stuken** [comment]
Spotted a couple of things.

Some of the settings icons are offset by a few pixels at 150% windows scaling.
Screenshot 2022-08-22 163113

Aslo switching to big picture via the toolbar then swapping back to the qt list or grid view via the view menu and booting a game/bios doesnt render the game in the window. Switching to fullscreen and back will correct it.

File: pcsx2/MTGS.cpp [hunk 1]
```diff
@@ -296,9 +298,23 @@ void SysMtgsThread::MainLoop()
 		// is very optimized (only 1 instruction test in most cases), so no point in trying
 		// to avoid it.
 
+#ifdef PCSX2_CORE
+		if (m_run_idle_flag.load(std::memory_order_acquire) && VMManager::GetState() != VMState::Running)
```
Comment (tellowkrinkle) [inline pcsx2/MTGS.cpp:302]: Nothing seems to store this flag with `memory_order_release`.  Is that a future or past plan?

Comment (stenzek) [reply pcsx2/MTGS.cpp:302]: Whoops, nice catch! SetRunIdle() should indeed be release here.

File: pcsx2/Frontend/ImGuiFullscreen.cpp [hunk 2]
```diff
@@ -0,0 +1,2164 @@
...
+		else if (v->has_value() && !v->value())
+			v->reset();
+		else
+			*v = true;
+	}
+
+	float t = v->has_value() ? (v->value() ? 1.0f : 0.0f) : 0.5f;
+	ImDrawList* dl = ImGui::GetWindowDrawList();
+	ImGuiContext& g = *GImGui;
+	float ANIM_SPEED = 0.08f;
```
Comment (lightningterror) [inline pcsx2/Frontend/ImGuiFullscreen.cpp:1127]: Would be nice to move it in the function since the check is commented out.

**Comment by Daisouji** [comment]
It's working.  If you go from grid/list to BP, and you can back to BP with System Display.

**Review by Mrlinkwii — APPROVED** [review]

**Review by JordanTheToaster — APPROVED** [review]

**Comment by Daisouji** [comment]
Found 2 things :

-   Interacting with anything in IMGUI menu while using -nogui and -bigpicture won't allow you to exit the program
-   The render with "seperate window" option w/ -bigpicture command gives the same results

**Pull request merged**